src/snowflake/snowpark/mock/plan.py

```diff
@@ -708,7 +718,13 @@ def calculate_expression(
             # expr_id maps to the projected name, but input_data might still have the exp.name
             # dealing with the KeyError here, this happens in case df.union(df)
             # TODO: check SNOW-831880 for more context
-            return input_data[exp.name]
+            try:
+                # TODO: quoting is inconsistent during calculation, some columns are quoted while some are not
+                #  we need to decouple the logic between intermediate calculation and result representation,
+                #  check SNOW-942397 for more context
```

Comment:
This sounds very error prone, should we aim to address this in PuPr instead?

Reply:
it is error prone.. the two logic are intertwined /facepalm/
I will see what I can do to improve

Reply:
Thanks for looking into it Adam! If you find it to be too much refactoring, I'm happy to keep it in post-PuPr.

Reply:
actually that's an issue in the aggregation where we set the column name, I found that we shall just add quotes for the intermediate results, such that quotes are aligned within our code.

thanks for pushing this

Reply:
updated to use `calculate_expression` to literal computation and single quote literal all the time in analyze